plugins/pulp_python/plugins/models.py

```diff
@@ -3,23 +3,58 @@
 import re
 import tarfile
 
+from mongoengine import StringField
+from pulp.server.db.model import FileContentUnit
+
 from pulp_python.common import constants
 
 
 DEFAULT_CHECKSUM_TYPE = 'sha512'
 
 
-class Package(object):
+# These are required to be in the PKG-INFO file.
+REQUIRED_ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
+                  'description', 'platform')
+
+
+class Package(FileContentUnit):
     """
     This class represents a Python package.
     """
 
-    TYPE = constants.PACKAGE_TYPE_ID
-    # The full list of supported attributes. Attributes beginning with underscore are specific to
-    # this module and are not found in PKG-INFO.
-    _ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
-              'description', 'platform', '_filename', '_checksum', '_checksum_type',
-              '_metadata_file')
+    # unit key
+    name = StringField(required=True)
+    version = StringField(required=True)
+
+    author = StringField()
+    author_email = StringField()
+    description = StringField()
+    home_page = StringField()
+    license = StringField()
+    platform = StringField()
+    summary = StringField()
+
+    _checksum = StringField()
```

Comment:
I actually like the idea that a new checksum should only be calculated deliberately, hopefully right after the content has been verified some other way. In the python package case, we verify with an md5 checksum, then calculate a new sha512 checksum.

It would be a shame to have a situation where there is no checksum provided, the file is somehow corrupt, and we end up automatically saving a new checksum of the corrupt file without realizing it.